src/Channels.Networking.TLS/SecureChannel.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Channels.Networking.TLS
+{
+    public class SecureChannel: IChannel
+    {
+        IWritableChannel _lowerOutputChannel;
+        IReadableChannel _lowerInputChannel;
+
+        Channel _outputChannel;
+        Channel _inputChannel;
+
+        public IReadableChannel Input => _outputChannel;
+        public IWritableChannel Output => _inputChannel;
+
+        public SecureChannel(IChannel inChannel, ChannelFactory channelFactory)
+            :this(inChannel.Output, inChannel.Input,channelFactory)
+        {
+        }
+
+        public SecureChannel(IWritableChannel outputChannel, IReadableChannel inputChannel, ChannelFactory channelFactory)
+        {
+            _lowerInputChannel = inputChannel;
+            _lowerOutputChannel = outputChannel;
+            _inputChannel = channelFactory.CreateChannel();
+            _outputChannel = channelFactory.CreateChannel();
+        }
+
+        internal async void StartReading<T>(T securityContext) where T : ISecureContext
+        {
+            try
+            {
+                var token = securityContext.ProcessContextMessage(default(ReadableBuffer));
+                if (token != null)
+                {
+                    var writeBuff = _lowerOutputChannel.Alloc(token.Length);
+                    writeBuff.Write(token);
+                    await writeBuff.FlushAsync();
+                }
+
+                while (true)
```

Comment:
This channel consuming loop needs to be fixed to look like this https://github.com/davidfowl/Channels/blob/master/samples/Channels.Samples/Framing/Codec.cs#L32-L67